federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -638,7 +638,7 @@ func (s *ServiceController) updateAllServicesToCluster(services []*cachedService
 	cluster, ok := s.clusterCache.clientMap[clusterName]
 	if ok {
 		for _, cachedService := range services {
-			appliedState := cachedService.appliedState
+			appliedState := cachedService.lastState
```

Comment:
same. can you explain why this change?
Will also be good to add a comment here in code explaining why we are using lastState and not appliedState.